src/SIL.LCModel/DomainServices/DataMigration/DataMigration7000072.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) 2016-2017 SIL International
+// This software is licensed under the LGPL, version 2.1 or later
+// (http://www.gnu.org/licenses/lgpl-2.1.html)
+
+using System.Xml.Linq;
+
+namespace SIL.LCModel.DomainServices.DataMigration
+{
+	/// ----------------------------------------------------------------------------------------
+	/// <summary>
+	/// Migrate data from 7000071 to 7000072.
+	/// </summary>
+	/// ----------------------------------------------------------------------------------------
+	internal class DataMigration7000072 : IDataMigration
+	{
+		/// <summary>
+		/// 1) Add the reference collection of Senses to each reversal index and
+		/// fill it with the Senses that referenced the ReversalIndexEntry
+		/// 2) Remove the collection of ReversalIndexEntry from each Sense
+		/// 3) Migrate any VirtualOrdering objects if necessary
+		/// </summary>
+		/// <param name="repoDto"></param>
+		public void PerformMigration(IDomainObjectDTORepository repoDto)
+		{
+			DataMigrationServices.CheckVersionNumber(repoDto, 7000071);
+
+			AddSensesToReversalIndexEntry(repoDto);
+
+			RemoveReversalEntriesFromSenses(repoDto);
+
+			ChangeReferringSensesToSenses(repoDto);
+
+			DataMigrationServices.IncrementVersionNumber(repoDto);
+		}
+
+		/// <summary>
+		/// Add the reference collection of Senses to each reversal index
+		/// and fill it with the Senses that referenced the ReversalIndexEntry
+		/// </summary>
+		/// <param name="repoDto"></param>
+		public void AddSensesToReversalIndexEntry(IDomainObjectDTORepository repoDto)
+		{
+			// Get all the LexSense Classes
+			var allLexSenses = repoDto.AllInstancesWithSubclasses("LexSense");
+			foreach (var aLexSense in allLexSenses)
+			{
+				var lexSenseElement = XElement.Parse(aLexSense.Xml);
+				// Get the Current LexSense's Guid
+				var lexSenseGuid = lexSenseElement.Attribute("guid")?.Value;
+
+				// Get the ReversalEntries within the LexSense
+				var reversalEntries = lexSenseElement.Element("ReversalEntries");
+				if (reversalEntries == null)
+					break;
+				var reversalEntriesList = reversalEntries.Elements("objsur");
+
+				// Loop through the ReversalEntries within the LexSense
+				foreach (var aReversalEntry in reversalEntriesList)
+				{
+					// Get the Current ReversalEntries (objsur) Guid from the objsur
+					var reversalEntryGuid = aReversalEntry.Attribute("guid")?.Value;
+
+					// Get the Reversal Entry Object, that has the specified Guid, that needs to be modified
+					var aReversalIndexEntry = repoDto.GetDTO(reversalEntryGuid);
+
+					// Add Senses to the Reversal Entry
+					var reversalIndexElement = XElement.Parse(aReversalIndexEntry.Xml);
+					var sensesElement = reversalIndexElement.Element("Senses");
+					if (sensesElement == null)
+					{
+						XElement newSensesElement = new XElement("Senses",
+							new XElement("objsur", new XAttribute("guid", lexSenseGuid),
+								new XAttribute("t", "o")));
```

Comment:
The attribute "t" should be set to "r" here and in the else below. "o" means owning, so if the ReversalIndexEntry was removed it would remove the Sense. "r" means reference so when the ReversalIndexEntry is removed the Sense will remain.